src/core/libmaven/PeakGroup.cpp

```diff
@@ -203,12 +197,50 @@ bool PeakGroup::isMS1()
     return false;
 }
 
+bool PeakGroup::hasCompoundLink() const
+{
+    if(hasSlice() && _slice.compound != NULL)
+        return true;
+
+    return false;
+}
+
+Compound* PeakGroup::getCompound()
+{
+    if (hasSlice()) {
+        return _slice.compound;
+    }
+    return NULL;
+}
+
+void PeakGroup::setCompound(Compound* compound)
```

Comment:
Would be safer to check whether a slice exists before setting a compound.

Reply:
An empty slice (having zeros as its bounds) is the default state. So there is no chance of an invalid assignment (that leads to a crash).